packages/todo-common-lib/src/infra/mysql/todo/repositories/to_do.repository.ts

```diff
@@ -0,0 +1,79 @@
+import { Cursor, FindOptions, In } from 'nextbeat-lib/infra/mysql';
+import { Todo } from 'todo-common/domain';
+import { TodoDB } from '../datasource';
+import { ToDoTable } from '../tables';
+import { RepositoryInterface } from './repository.interface';
+
+export class ToDoRepository implements RepositoryInterface {
+  /**
+   * IDで抽出する
+   * @param id
+   */
+  async findById(id: number): Promise<ToDoTable | undefined> {
+    return TodoDB().findOne(ToDoTable, { where: { id } }, 'replica');
+  }
+
+  /**
+   * ステータスで抽出する
+   * @param criteria
+   * @returns
+   */
+  async findByStates(
+    criteria: ToDoRepository.FilterCondition,
+  ): Promise<ToDoTable[] | undefined> {
+    let where: FindOptions<ToDoTable>['where'];
+
+    const condition = criteria.byStates;
+    if (condition.states && condition.states?.length > 0) {
```

Comment:
検査済みなので optional chain は不要だと思います。

```suggestion
    if (condition.states && condition.states.length > 0) {
```